### PR TITLE
add initial implementation

### PR DESCRIPTION
**Pull request by hectorj2f**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module github.com/hectorj2f/cargobump
```
Comment (cpanato) [inline go.mod:1]: need to use chainguard-dev

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+go 1.22.6
```
Comment (cpanato) [inline go.mod:3]: can we update to 1.23?

File: pkg/parser/testdata/bumpfile.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+packages:
+  - name: name-1
+    version: version-1
+  - name: name-2
+    version: version-2
+  - name: name-3
+    version: version-3
```
Comment (cpanato) [inline pkg/parser/testdata/bumpfile.yaml:7]: nit: new line

File: pkg/parser/parser.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
+package parser
```
Comment (cpanato) [inline pkg/parser/parser.go:1]: add the header boilerplate

**Review by cpanato — CHANGES_REQUESTED** [review]

File: pkg/parser/parser.go (hunk 4 above)
Comment (hectorj2f) [reply pkg/parser/parser.go:1]: I checked pombump and infra-images. We didn't add the boilerplate :).

Comment (hectorj2f) [reply pkg/parser/parser.go:1]: Maybe we can move on without.

Comment (cpanato) [reply pkg/parser/parser.go:1]: ok no worries i can do in a follow up

**Comment by cpanato** [comment]
@hectorj2f ci is not happy

**Comment by hectorj2f** [comment]
CI is now happy 😊!

**Comment by hectorj2f** [comment]
@cpanato could you have another look?

**Review by cpanato — APPROVED** [review]

**Comment by hectorj2f** [comment]
Thanks @cpanato 🎉🎉

**Pull request merged**